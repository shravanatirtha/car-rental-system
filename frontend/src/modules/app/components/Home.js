import React, { useState, useEffect } from "react";
import "./App.css";
import { config } from "../../../config/constants.js";

const Home = () => {
  const [message, setMessage] = useState("");

  useEffect(() => {
    fetch(config.BASE_PATH + "/hello")
      .then((response) => response.text())
      .then((error) => {
        setMessage(error);
      });
  }, []);

  return (
    <div className="App">
      <header className="App-header">
        <img
          src={process.env.PUBLIC_URL + "/assets/logo.svg"}
          className="App-logo"
          alt="logo"
        />
        <h1 className="App-title">{message}</h1>
      </header>
      <p className="App-intro">
        To get started, edit <code>src/App.js</code> and save to reload.
      </p>
    </div>
  );
};

export default Home;
