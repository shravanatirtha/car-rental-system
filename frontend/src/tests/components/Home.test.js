import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../../modules/app/components/Home';

it('Home renders correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});