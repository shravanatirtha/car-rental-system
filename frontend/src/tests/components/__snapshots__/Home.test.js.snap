// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="/assets/logo.svg"
    />
    <h1
      className="App-title"
    >
      
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
